### ISSUE TITLE
Add Scotland to Freecen2 Districts; Pieces; Civil Parishes

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/FreeCENMigration#1060]

**Comment by Captainkirkdawson** [comment]
The issue is not so much CSVProc as getting the Place/District/Piece/Parish parameter information for Scotland like we have from TNA Without it we cannot proceed

**Comment by PatReynolds** [comment]
Emailed contact at NLS 4 Jan 2020

**Comment by PatReynolds** [comment]
Contact at NLS has sent files: sent on to Geoff and Rhoda, in FreeCEN>Data Documentation folder.

**Comment by Captainkirkdawson** [comment]
Geoff replied
This looks like part of the structure for a Scotland PARMS to me. It had RDs, piece numbers and counties. It would need another table for a join of piece names and civil parishes wityh a join field of piece numbers. It may be worth having a saved copy as a start.

Unfortunately it does not paint the whole picture.

**Comment by PatReynolds** [comment]
Hi @geoffj-FUG , so you have the RDS, piece numbers and counties - but need the numbers, piece names and civil parishes as a separate table (just checking before I ask for the wrong thing!)

**Comment by geoffj-FUG** [comment]
Pat

 

Kirk knows the exact file structure so I have sent him a copy of this.

 

My comment re a separate table was not a specific request but a means of
joining 2 sets of data. The important thing is we need to relate civil
parishes to piece numbers. I do not think that there were civil parishes in
the download.

 

Kirk would be the expert on the exact file structure as I do not have a copy
of the data table.

 

Geoff

**Comment by Captainkirkdawson** [comment]
I have reviewed the Scotland data structure provided as well as the freecen1 piece structure and concluded the following.
Scotland has a different structure to England and Wales. 
With the exception of 1841 England and Wales have no more than 1 district for a piece and a district may have multiple pieces. Each piece can have 1 or more civil parishes. In addition to District and Civil Parish names Pieces are normally associate with a name with the piece number.
Scotland can have multiple districts in a single piece number and a district can have multiple Civil Parishes and they do not assign a name to a piece. 
The new information from the Scotland contact shows that while the basic structure remains similar they are creating a new District number system. 
Transcriptions for Scotland by Freecen have been done and loaded into FC1 at the Civil Parish level by incorporating a unique parish number in the file name and can be split in many blocks. Each block with a separate file name. Civil Parishes have unique names within the district and basic searching is done by the district name.
Until such a time as we recommence Scottish transcriptions there seems no point in trying to convert to the new piece number.
Hence I would propose defining a district/piece/parish structure for FC2 based  on the freecen_piece information from FC1

**Comment by Captainkirkdawson** [comment]
@FreecenBren @geoffj-FUG @PatReynolds @DeniseColbert 
I have developed Freecen2 data structures of Place, District, Piece and Civil Parish for Scotland ans deployed to test3 for review. There are no active VLD file connections at this point as we need to look at the Place linkages; I would suggest these not be fixed on test3 but on production. (This code could be deployed to production with no effect on its performance). The displays indicate a number of spelling and related issues that also need to be addressed; but again only in production. 
You are asked to review and consider.
If we move on this we can also move on the full transition to vld usage of the gazetteer place names in the census district searches.

**Comment by Captainkirkdawson** [comment]
@rhodamackenzie I forgot to add you to the list for review

**Comment by rhodamackenzie** [comment]
I have not previously looked at higher order Scotland census details. In test3 I looked at ABD. 1841-71 listed. I'm seeing that district names and piece names are identical - i.e. there are 87 of each, same list. There are 132 civil parishes listed here. The additional names not identical to the 87 district and pieces are almost all cross border parishes and inner city/large town areas. 

Cross county parishes have the county code after them for the OTHER county so presumably need moved out leaving even fewer parishes. 

AYR has 1841-51. 45 identical districts and pieces and 49 parishes, again identical list but with 4 extras. 

I suppose it doesn't really matter if district, piece and parish names are the same, but it will be a palaver dealing with multiple districts for one piece unless we swap the hierarchy and put piece in at the top of the tree, then district, then parish. 

It's very tempting to drop the 3 level system and either use only pieces and CPs, particularly as a single piece can have several districts, or just district and CP, or even (gasp...) have only civil parishes.

**Comment by Captainkirkdawson** [comment]
Thanks Rhoda

I agree it's a little bit of a cludge. However we must have a consistent
data model for all the countries; we cannot have different structures for
each country. Perhaps the original design should have been a many to many
between District and Piece. But we didn't and it would be an enormous job
to work that in.

Really there are only 2 levels .But to make the data model function the
District and Piece are essentially the same apart from the piece having a
piece number.

Lets recall that the search is currently based on the District and the
desire is to add a search at the Parish level so this structure will work.

Lets also recall that currently transcribing Scotland is inactive and these
are transcriptions done some years ago. The need for these FR2 entries is
so that we can move the whole database to the use of the Gazetteer places.
Using this we can create a few missing Scottish parishes and have an
integrated search.

.


Kirk Dawson
Fairmont, B.C. Canada

**Comment by FreecenBren** [comment]
Hi, 
 
Not sure what contribution I can make.


I used to have issues just getting them uploaded if the unique VLD number as below, was wrong.
 
Each PIECE has an extra ‘digit’  reference to each VLD. A Piece can have more than 1 VLD.

Rather like doing the England and Wales Pieces by each ED number, instead of all being  joined together to make the one Piece.
See 3rd Column below.


This example Piece number SCT1841/622  has 25 parts as per the VLD  list following. For 622.
In front of the VLD number itself  and after the census year they added  either a 0 to a 9.

If they wanted more numbers  they then started again at zero and changed the Census year from example 41 to 42 etc.
The number I have added in the last column is the VLD number we uploaded each VLD part using the above example.
 
Example SCT1841/622  LKS
 
https://freecen1.freecen.org.uk/coverage-LKS.html#1841
 
Lanarkshire LKS 1841 has 25 VLD’s for this one PIECE number of SCT1841/622
 
	
 	

Piece
Place
Comprising
Status
VLD number

SCT1841/621
Avondale
Avondale
On-Line
HS410621

SCT1841/622
Barony
Anderston 1-12
On-Line
HS411622
	
Barony
Anderston 13-23
On-Line
HS412622
	
Barony
Anderston 24-34
On-Line
HS413622
	
Barony
Barony Institutions
On-Line
HS414622
	
Barony
Blythswood 1-7
On-Line
HS415622
	
Barony
Blythswood 19-29
On-Line
HS416622
	
Barony
Blythswood 30-42
On-Line
HS417622
	
Barony
Blythswood 8-18
On-Line
HS418622
	
Barony
Bridgeton 1-6
On-Line
HS419622
	
Barony
Bridgeton 14-25
On-Line
HS421622
	
Barony
Bridgeton 7-13
On-Line
HS422622
	
Barony
Calton 1-10
On-Line
HS423622
	
Barony
Calton 11-23
On-Line
HS424622
	
Barony
Calton 24-38
On-Line
HS425622
	
Barony
Calton 39-59
On-Line
HS426622
	
Barony
Calton 60-71
On-Line
HS427622
	
Barony
Eastern 1-8
On-Line
HS428622
	
Barony
Eastern 15-22
On-Line
HS429622
	
Barony
Eastern 9-14
On-Line
HS431622
	
Barony
Finnieston 1-4
On-Line
HS432622
	
Barony
Maryhill
On-Line
HS433622
	
Barony
Milton 1-6
On-Line
HS434622
	
Barony
Milton 14-22
On-Line
HS435622
	
Barony
Milton 7-13
On-Line
HS436622
	
Barony
North & Eastern
On-Line
HS437622

SCT1841/623
Biggar
Biggar
On-Line
HS410623
 
Hope that makes sense.
 
I do look after the researchers queries for all of Scotland (With Margaret’s Help)
As I do not have the images I send the query to Margaret and she then sends be an amended  VLD if she had to amend any.


Margaret does not have all the images only some of them. She does have an account with ‘Scotland People’ so that helps.
 
The reason being as when Scotland COORD’s completed a County they returned the images  back to the Scotland Archives.
I say ‘Images’ but many were on ‘Micro Film or Tape’
 
So basically that is all I know about Scotland and you all  might already know that.
Well I do have relatives who come from Scotland but that means nothing as my father was Welsh!
 
So, I will wait  for more from whoever, as to what you  have assigned me to do.
Of course that also depends if I know it that is. 
 
Talk soon.
 
Brenda

**Comment by Captainkirkdawson** [comment]
Thank you @FreecenBren That was an extremely good example.
The FC2 structure for that example shows what you would have expected at the district level.
![lks1.png](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/78de6f3a-00cd-4f8a-8a47-e0d2f20665fa)
And again at the Piece level
![lks2.png](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/c4cdf3a7-56bd-44da-849b-d5b180ca6d54)

**Comment by Captainkirkdawson** [comment]
Structure still to be reviewed by @geoffj-FUG when he is available..
Propose to run the set up in production without making the entries so as to identify the missing place name. @rhodamackenzie has agreed to review and create as required.
f2rake create_freecen2_parms_scotland[5010] --trace

**Comment by geoffj-FUG** [comment]
Kirk, Rhoda

 

We developed the Scotland database to the point that we had spreadsheets for each year. These were reviewed by Rhoda as if my memory is correct.

 

I have attached the spreadsheets (I have cc’d to Kirk and Rhoda so that they actually get them).

 

We did not develop testing for the fields that are not already in England and Wales and Channel Islands. They are not marked as tested on my record.

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I had given little or no thought about the actual CSV processing of Scotland records as from what I understand they are inactive.
The main purpose of the work to date is to be able to support Scottish searches using the Freecen2Place names instead of the Place

**Comment by Captainkirkdawson** [comment]
Revised the title to more correctly reflect the current work.
Cose deployed to test3 and Gazetteer names added to production in readiness for deployment

**Comment by DeniseColbert** [comment]
Rhoda says this is coming along, she's checked this over.
